beta-src/src/utils/formatTime.ts

```diff
@@ -0,0 +1,21 @@
+import { ParsedTime } from "../interfaces/ParsedTime";
+
+export default function formatTime(time: ParsedTime) {
+  const availableTimeIntervals = Object.keys(time).filter((key) => time[key]);
+  const timeIntervalAbbreviations = availableTimeIntervals.map(
+    (interval: string) => {
+      return interval.charAt(0);
+    },
+  );
```

Comment:
You can define the `ParsedTime` interface as follows: 
```
interface ParsedTime {
    d: number;
    h: number;
    m: number;
    s: number;
}
```
This way you don't have to do lines 5-9. Additionally, it makes the result of line 4 re-usable.